jetson/percep/pcl.cpp

```diff
@@ -160,25 +160,60 @@ void PCL::FindInterestPoints(std::vector<pcl::PointIndices> &cluster_indices,
 
         //Initialize interest points
         std::fill(curr_cluster->begin(), curr_cluster->end(), cluster_indices[i].indices[0]);
-        
-        //Order of interest points: 0=Up Left 1=Up Right 2=Low Right 3=Low Left
+
+        //Interest Points: 0=Leftmost Point 1=Rightmost Point 2=Lowest Point 3=Highest Point 4=Closest Point 5=Furthest Point
         for (auto index : cluster_indices[i].indices)
         {
             auto curr_point = pt_cloud_ptr->points[index];
             
-            //Upper Left
-            if(curr_point.x < pt_cloud_ptr->points[curr_cluster->at(0)].x && curr_point.y > pt_cloud_ptr->points[curr_cluster->at(0)].y)
+            if(curr_point.x < pt_cloud_ptr->points[curr_cluster->at(0)].x){
                 curr_cluster->at(0) = index;
-            //Upper Right
-            else if(curr_point.x > pt_cloud_ptr->points[curr_cluster->at(1)].x && curr_point.y > pt_cloud_ptr->points[curr_cluster->at(1)].y)
+            }
+            if(curr_point.x > pt_cloud_ptr->points[curr_cluster->at(1)].x){
                 curr_cluster->at(1) = index;
-            //Low Left
-            else if(curr_point.x < pt_cloud_ptr->points[curr_cluster->at(2)].x && curr_point.y < pt_cloud_ptr->points[curr_cluster->at(2)].y)
+            }
+            if(curr_point.y < pt_cloud_ptr->points[curr_cluster->at(2)].y){
                 curr_cluster->at(2) = index;
-            //Low Right
-            else if(curr_point.x > pt_cloud_ptr->points[curr_cluster->at(3)].x && curr_point.y < pt_cloud_ptr->points[curr_cluster->at(3)].y)
+            }
+            if(curr_point.y > pt_cloud_ptr->points[curr_cluster->at(3)].y){
                 curr_cluster->at(3) = index;
+            }
+            if(curr_point.z < pt_cloud_ptr->points[curr_cluster->at(4)].z){
+                curr_cluster->at(4) = index;
+            }
+            if(curr_point.z > pt_cloud_ptr->points[curr_cluster->at(5)].z){
+                curr_cluster->at(5) = index;
+            }
+        }
+
+        //Calulates the width of the obstacle based the difference between the leftmost and rightmost interst point
+        double width = std::abs(pt_cloud_ptr->points[curr_cluster->at(1)].x - pt_cloud_ptr->points[curr_cluster->at(0)].x);
+        //Calculates the number of rover widths that fit within the obstacle. The x10 multiplier adds more width increments.
+        int rovers = ((int) width/ROVER_W_MM) * 10;
```

Comment:
Why did you use 10 here? Could we get away with a smaller number, so we have to deal with fewer points?

Reply:
If I make it less than 10, then parts of the hill are not detected as obstacles in test1. I could add back the depth points which I deleted earlier and see if it fixes the issue.

Reply:
Just a nitpick here, the variable name "rovers" isn't particularly self explanatory. If it is not common vernacular within the perception team I would suggest a refactor.

Reply:
I changed the variable name increments